Research/ResearchUI/iOS/Step View Controllers/RSDScrollingOverviewStepViewController.swift

```diff
@@ -141,58 +139,279 @@ open class RSDScrollingOverviewStepViewController: RSDOverviewStepViewController
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return self.iconCollectionView.gridLayout.cellSize(for: indexPath)
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        return self.iconCollectionView.gridLayout.seciontInset(for: section)
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+
+        let cell = self.iconCollectionView.dequeueReusableCell(withReuseIdentifier: iconCollectionViewCellResuableCellId, for: indexPath)
+        
+        // The grid layout stores items as (section, row),
+        // so make sure we use the grid layout to get the correct item index.
+        let itemIndex = self.iconCollectionView.gridLayout.itemIndex(for: indexPath)
+
+        if let overviewCell = cell as? RSDOverviewCollectionViewCell {
+            overviewCell.indexPath = indexPath
+            overviewCell.setDesignSystem(self.designSystem, with: self.backgroundColor(for: .body))
+            if let icons = self.overviewStep?.icons,
+                indexPath.row < icons.count {
+                let icon = icons[itemIndex]
+                overviewCell.imageView?.image = icon.icon?.embeddedImage()
+                overviewCell.titleLabel?.text = icon.title
+                overviewCell.setNeedsUpdateConstraints()
+            }
+        }
+
+        return cell
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        // Only have a section header in the first section.
+        if section == 0 {
+            return CGSize(width: iconCollectionView.bounds.width, height: self.iconCollectionViewHeaderHeight)
+        }
+        // Otherwise make the size of the section have no height.
+        return CGSize(width: iconCollectionView.bounds.width, height: 0)
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        
+        if kind == UICollectionView.elementKindSectionHeader {
+            let headerCell = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: iconCollectionHeaderTitleCellResuableCellId, for: indexPath)
+
+            if let rsdCell = headerCell as? RSDTitleHeaderCollectionViewHeader {
+                rsdCell.setDesignSystem(self.designSystem, with: self.backgroundColor(for: .header))
+                rsdCell.titleLabel?.text = self.iconCollectionViewHeaderTitle
+            }
+            
+            return headerCell
+        }
+        return UICollectionReusableView()
```

Comment:
```
else {
    return UICollectionReusableView()
}
```

Be explicit.

Alternatively, you could do this:

```
guard kind == UICollectionView.elementKindSectionHeader 
    else { 
        return UICollectionReusableView() 
}
```

where the early exit is explicitly the "if not" case that is really what you are testing for and means that you don't have to wrap the entire func code in an extra set of curly brackets.

Reply:
FTFY